SYCL/ESIMD/api/functional/type_coverage.hpp

```diff
@@ -86,28 +91,62 @@ template <tested_types required> auto get_tested_types() {
   }
 }
 
+enum class tested_dims { all };
+
+// Factory method to retrieve pre-defined values_pack, to have the same
+// default dimensions over the tests
+template <tested_dims required> auto get_tested_dimensions() {
+  if constexpr (required == tested_dims::all) {
+    return values_pack<1, 8, 16, 32>();
+  } else {
+    static_assert(required != required, "Unexpected tested dimension");
+  }
+}
+
 // Run action for each of types given by named_type_pack instance
-template <template <typename, typename...> class action,
-          typename... actionArgsT, typename... types, typename... argsT>
-inline bool for_all_types(const named_type_pack<types...> &typeList,
-                          argsT &&... args) {
+template <template <typename, int, typename...> class Action, int N,
+          typename... ActionArgsT, typename... Types, typename... ArgsT>
+inline bool for_all_types(const named_type_pack<Types...> &type_list,
+                          ArgsT &&... args) {
   bool passed{true};
 
   // run action for each type from types... parameter pack
-  size_t typeNameIndex = 0;
-
-  int packExpansion[] = {
-      (passed &= action<types, actionArgsT...>{}(std::forward<argsT>(args)...,
-                                                 typeList.names[typeNameIndex]),
-       ++typeNameIndex,
-       0 // Dummy initialization value
-       )...};
+  size_t type_name_index = 0;
+
+  ((passed &= Action<Types, N, ActionArgsT...>{}(
+        std::forward<ArgsT>(args)..., type_list.names[type_name_index]),
+    ++type_name_index),
+   ...);
+  // Every initializer clause is sequenced before any initializer clause that
+  // follows it in the braced-init-list. Every expression in comma operator is
+  // also strictly sequenced. So we can use increment safely. We still should
+  // discard dummy results, but this initialization should not be optimized out
+  // due side-effects
+
+  return passed;
+}
+
+// Calls for_all_types for each vector length by values_pack instance
+template <template <typename, int, typename...> class Action,
+          typename... ActionArgsT, typename... Types, int... Dims,
+          typename... ArgsT>
+inline bool for_all_types_and_dims(const named_type_pack<Types...> &type_list,
+                                   const values_pack<Dims...> &intList,
+                                   ArgsT &&... args) {
+  bool passed{true};
+
+  // run action for each value from values_pack... parameter pack
+  int int_index = 0;
+
+  ((passed &= for_all_types<Action, Dims, ActionArgsT...>(
+        type_list, std::forward<ArgsT>(args)...),
+    ++int_index),
+   ...);
   // Every initializer clause is sequenced before any initializer clause that
   // follows it in the braced-init-list. Every expression in comma operator is
   // also strictly sequenced. So we can use increment safely. We still should
   // discard dummy results, but this initialization should not be optimized out
   // due side-effects
```

Comment:
```suggestion
  // Run action for each value from values_pack... parameter pack
  ((passed &= for_all_types<Action, Dims, ActionArgsT...>(
        type_list, std::forward<ArgsT>(args)...)),
   ...);
  // The unary right fold expression is used for parameter pack expansion.
  // An additional pair of brackets is required because of precedence of any
  // operator relatively to the comma operator.
```

Reply:
The comments was replaced in this commit: 9ce01972c9d4b14c835faf126b6913a107449067